src/lib/layout/aem-remote/aem-remote.component.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'aem-remote-component',
+  templateUrl: './aem-remote.component.html'
+})
+
+export class AEMRemoteComponent {
```

Comment:
Shouldn't this component inherit from something?

Reply:
Not necessary. This component is mostly acting as a placeholder so that the customer can avoid having to include the model-provider logic for each and every new component being created. The app can simply use the selector `aem-remote-component` and pass in the pagePath and itemPath and AEM-specific logic for model provider gets added in here.